bwt/wavelet.go

```diff
@@ -0,0 +1,340 @@
+package bwt
+
+import (
+	"fmt"
+	"math"
+
+	"golang.org/x/exp/slices"
+)
+
+// waveletTree datastructure that allows us to
+// conduct RSA queries on strings.
```

Comment:
I know it's definitely spelled out somewhere in this PR but for newbies (and me!) it'd be good to have the abbreviation spelled out in parentheses the first time it appears in a file. What does RSA stand for again?

Reply:
Addressed, LMK if that's better.